### ISSUE TITLE
Add option to runIde using a IJ-community source

### ISSUE THREAD

**Issue by brian-mcnamara** [JetBrains/intellij-platform-gradle-plugin#628]
There are times when I need to make changes to IJ community and inspect the outcome from my plugin. It would be nice if I could point the plugin to the local IJ sources and have runIde use that as the runtime.

I think its fair to require the sources to be compiled before attempting to call runIde

**Comment by zolotov** [comment]
> I think its fair to require the sources to be compiled before attempting to call runIde

If IDE is already built, you can employ `runIde#ideDirectory` to point to the compiled distribution. I don't see what else can be added in `gradle-intellij-plugin` in that regard. I think, compilation of the `intellij-community` shouldn't be part of `gradle-intellij-plugin`

**Comment by brian-mcnamara** [comment]
Yes, creating the distribution and using ideDirectory is the workaround, but creating the distribution is another step outside a normal compile. I was hoping we could just point to the out/production path and run IJ through the compiled classpath. However just a thought depending on the complexity, I believe Yann mentioned this would be very complex in which case its not worth it.

**Comment by zolotov** [comment]
>  I was hoping we could just point to the out/production path and run IJ through the compiled classpath.

Got it. I’ll check how hard that would be to implement. Looks like the only problem is to read the project model and build a proper classpath out from it.

**Comment by hsz** [comment]
We're closing this as out of scope.

Compiling `intellij-community` (or synthesizing a run classpath from `out/production`) isn't something this plugin will do — that's the community project's own build.

If you already have a built distribution, 2.x can run against it:

```kotlin
dependencies {
    intellijPlatform {
        local("/path/to/built/idea")
    }
}
```

`local(...)` needs a real IDE layout with `product-info.json` and `lib/`, not a source checkout.

If running straight from compiled sources is still a hard requirement, please open a new issue with a current 2.x repro and why the distribution-based workflow isn't enough.
